src/backend/utils/adt/json.c

```diff
@@ -150,14 +152,28 @@ json_categorize_type(Oid typoid,
 	/* Look through any domain */
 	typoid = getBaseType(typoid);
 
+	if (sql_dialect==SQL_DIALECT_TSQL)
```

Comment:
In here, the result should depend on the type, not sql_dialect. So, please add this logic to the below switch-case. In other words, even in postgres dialect, if typoid is `sys.bit` then it should behave the same as in T-SQL.

Reply:
Resolving all the comments as just moving declaration from parser.h to gram-tsql-prologue.y.h